hw_11/HW_1_Sotnikov.py

```diff
@@ -0,0 +1,368 @@
+        if isinstance(attribute, list):
+            answers = []
+            for attr in attribute:
+                if isinstance(attr, str) and hasattr(self, attr):
+                    answers.append(f"{attr}: {getattr(self, attr)}")
+                else:
+                    raise AttributeError\
+                    ("You have entered unavailable argument.\n\
+                    Please input one of the below-writed attributes or list/tuple with them:\n\
+                    'user', 'date', 'time', 'message'")
+            return answers
+        elif isinstance(attribute, str) or isinstance(attribute, User):
+            if hasattr(self, attribute):
+                return f"{attribute}: {getattr(self, attribute)}"
+            else:
+                raise AttributeError\
+            ("You have entered unavailable argument.\n\
+            Please, input one of the below-writed attributes or list/tuple with them:\n\
+            'user', 'date', 'time', 'message'")
+        else:
+            raise AttributeError\
+            ("You have used unavailabel class of argument\n\
+            Please, use string type for message and User type for user.")
+
+    def send(self, chat):
+        if isinstance(chat, Chat):
+            self.datetime = datetime.today()     # Настраиваем дату и время
+            self.date = self.datetime.strftime("%d %m %Y")
+            self.time = self.datetime.strftime("%H:%M:%S")
+
+            chat.recieve(self)           # Передаем сообщение в чат
+
+# Creating class User
+class User:
+    def __init__(self, nickname, name="Bogdan", surname="Sotnikov", \
+                  birhdate="01.05.1999", sex="male",\
+                  user_status="user", what_is_he_doing="doing his python homework"):
+        self.name = name
+        self.surname = surname
+        self.nickname = nickname
+        self.birhdate = birhdate
+        self.sex = sex
+        self.user_status = user_status
+        self.what_is_he_doing = what_is_he_doing
+
+    def __repr__(self):
+        return str([self.nickname,
+                    self.name,
+                    self.surname,
+                    self.birhdate,
+                    self.sex,
+                    self.user_status,
+                    self.what_is_he_doing
+                   ])
+
+
+#2 Alien assignment
+class Args:
+    def __init__(self, *args, **kwargs):
+        self.argi = args
+        self.kwargi = kwargs
+
+    def __rlshift__(self, fun):
+        return fun(*self.argi, **self.kwargi)
+
+    def regeneration(self):
+        print("Восстановление нервных клеток")
+
+sum << Args([1, 2])
+
+(lambda a, b, c: a**2 + b + c) << Args(1, 2, c=50)
+
+
+#3 The successor
+
+class StrangeFloat(float):
+    def __init__(self, num):
+        super().__init__()
+        self.num = num
+        self.operands = {'add': lambda a, b: a + b, 'subtract': lambda a, b: a - b,
+                         'multiply': lambda a, b: a * b, 'divide': lambda a, b: a / b }
```

Comment:
Можно ещё так, чтобы не изобретать велосипед
```suggestion
        self.operands = {'add': self.__add__, 'subtract': lambda a, b: a - b,
                         'multiply': lambda a, b: a * b, 'divide': lambda a, b: a / b }
```